src/components/IncorporationAgreement/AgreementType.vue

```diff
@@ -1,82 +1,105 @@
+                <span>Undo</span>
+              </v-btn>
+            </v-flex>
+          </div>
+        </div>
+        <div v-else>
+          <v-card flat>
+            <v-radio-group v-model="agreementType" class="agreement-option-list">
+              <v-radio
+                v-for="(item, index) in incorporationAgreementTypeResource"
+                :key="index"
+                :value="item.code"
+                :id="`agreement-type-${item.code}`">
+                <template slot="label">
+                  <div v-html="item.description" class="agreement-option" />
+                </template>
+              </v-radio>
+            </v-radio-group>
+            <div class="action-btns">
+              <v-btn id="done-btn" large color="primary" @click="setAgreementType">
+                <span>Done</span>
+              </v-btn>
+              <v-btn id="cancel-btn" large outlined color="primary" @click="resetAgreementType">
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-card>
         </div>
-      </v-card>
-    </div>
-    <div v-else>
-      <v-card flat>
-        <v-radio-group v-model="agreementType" @change="changeAgreementType" class="agreement-option-list">
-          <v-radio v-for="(item, index) in incorporationAgreementTypeResource"
-           :key="index" :value="item.code" :id="`agreement-type-${item.code}`">
-            <template slot="label">
-              <div v-html="item.description" class="agreement-option"/>
-            </template>
-          </v-radio>
-        </v-radio-group>
       </v-card>
     </div>
   </div>
 </template>
 
 <script lang="ts">
 // Libraries
-import { Component, Vue, Prop } from 'vue-property-decorator'
-import { Action, State } from 'vuex-class'
+import { Component, Emit, Vue, Prop, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
 
 // Interfaces
-import { ActionBindingIF } from '@/interfaces'
+import { ActionBindingIF, IncorporationFilingIF } from '@/interfaces'
 import { AgreementTypeResource } from '@/resources'
 
 @Component
 export default class AgreementType extends Vue {
   private incorporationAgreementTypeResource = AgreementTypeResource
 
   // State
-  @State(state => state.stateModel.incorporationAgreementStep.agreementType)
-  readonly agreementTypeState: string | null
-
-  @Prop({ default: false })
-  private showErrorSummary: boolean
-
-  @Prop({ default: false })
-  private isSummary: boolean
+  @Getter getAgreementType!: string
+  @Getter getOriginalIA!: IncorporationFilingIF
 
   // Global setters
   @Action setIncorporationAgreementStepData!: ActionBindingIF
   @Action setHaveChanges!: ActionBindingIF
 
   // Local properties
   private agreementType: string | null = null
+  private showAgreementTypeForm = false
 
   // Lifecycle methods
   private created (): void {
-    this.agreementType = this.agreementTypeState
+    this.agreementType = this.getAgreementType
```

Comment:
Are you sure Agreement Type is set when this component is created? I'm not so sure.

I don't know if this is preferable, but you could set `{ immediate: true }` on your watcher below, which will cause it to fire when this component is created.

Reply:
I think it is a good idea to add immediate .. that way I dont need that setter in the created life cycle method